yacs/config.py

```diff
@@ -383,7 +389,7 @@ def _merge_a_into_b(a, b, root, key_list):
     for k, v_ in a.items():
         full_key = ".".join(key_list + [k])
```

Comment:
I think the logic in this loop has gotten a bit out of hand and L406 is too difficult to reason about. There are four cases to deal with as outlined in this simplified code sketch:

```python
if k in b:
    # Either recursively merge (if v is a CfgNode) or update in place
    if isinstance(v, CfgNode):
        _merge_a_into_b(v, b[k], root, key_list + [k])
    else:
        b[k] = v
else:
    # Either add new key inplace or raise an error (unless deprecated)
    if b.is_new_allowed():
        b[k] = v
    else:
        ...  # continue if deprecated else raise error
```

Before merging, I think it's necessary to refactor to make these cases more clear and remove the complicated logic.

Reply:
I have simplified logic, hope it is better now.

Reply:
Yes, this is great. Thanks!